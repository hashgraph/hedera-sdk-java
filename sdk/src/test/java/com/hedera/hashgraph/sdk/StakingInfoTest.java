package com.hedera.hashgraph.sdk;

import io.github.jsonSnapshot.SnapshotMatcher;
import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import java.time.Instant;

import static org.assertj.core.api.Assertions.assertThat;

public class StakingInfoTest {
    final Instant validStart = Instant.ofEpochSecond(1554158542);

    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    StakingInfo spawnStakingInfoAccountExample() {
        return new StakingInfo(
            true,
            validStart,
            Hbar.from(5),
            Hbar.from(10),
            AccountId.fromString("1.2.3"),
            null
        );
    }

    StakingInfo spawnStakingInfoNodeExample() {
        return new StakingInfo(
            true,
            validStart,
            Hbar.from(5),
            Hbar.from(10),
            null,
            3L
        );
    }

    @Test
    void shouldSerializeAccount() throws Exception {
        var originalStakingInfo = spawnStakingInfoAccountExample();
        byte[] stakingInfoBytes = originalStakingInfo.toBytes();
        var copyStakingInfo = StakingInfo.fromBytes(stakingInfoBytes);
        assertThat(copyStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", ""))
            .isEqualTo(originalStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", ""));
        SnapshotMatcher.expect(originalStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", "")).toMatchSnapshot();
    }

    @Test
    void shouldSerializeNode() throws Exception {
        var originalStakingInfo = spawnStakingInfoNodeExample();
        byte[] stakingInfoBytes = originalStakingInfo.toBytes();
        var copyStakingInfo = StakingInfo.fromBytes(stakingInfoBytes);
        assertThat(copyStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", ""))
            .isEqualTo(originalStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", ""));
        SnapshotMatcher.expect(originalStakingInfo.toString().replaceAll("@[A-Za-z0-9]+", "")).toMatchSnapshot();
    }
}
