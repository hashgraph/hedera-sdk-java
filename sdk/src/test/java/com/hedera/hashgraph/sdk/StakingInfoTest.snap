com.hedera.hashgraph.sdk.StakingInfoTest.shouldSerializeAccount=[
  "StakingInfo{declineStakingReward=true, stakePeriodStart=2019-04-01T22:42:22Z, pendingReward=5 ℏ, stakedToMe=10 ℏ, stakedAccountId=1.2.3, stakedNodeId=null}"
]


com.hedera.hashgraph.sdk.StakingInfoTest.shouldSerializeNode=[
  "StakingInfo{declineStakingReward=true, stakePeriodStart=2019-04-01T22:42:22Z, pendingReward=5 ℏ, stakedToMe=10 ℏ, stakedAccountId=null, stakedNodeId=3}"
]
